src/zarr/storage/_logging.py

```diff
@@ -18,8 +19,10 @@
 
     counter: defaultdict[str, int]
 
+T_Store = TypeVar("T_Store", bound=Store)
 
-class LoggingStore(WrapperStore[Store]):
+
+class LoggingStore(WrapperStore[T_Store]):
```

Comment:
Could this get a changelog entry mentioning `LoggingStore` is now a generic class?